integreat_cms/static/src/js/unsaved-warning.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * This file contains a function to warn the user when they leave a content without saving
+ */
+
+let confirmed = false;
+let edited = false;
+
+
+window.addEventListener("beforeunload", (event) => {
+  // trigger only when something is edited and no submit/save button clicked
+  if (edited && !confirmed) {
+    event.preventDefault();
+    event.returnValue = "This content is not saved. Would you leave the page?";
+  }
+});
+
+
+window.addEventListener("load", () => {
+  const form = document.querySelector("[data-unsaved-warning]");
+  // checks whether the user typed something in the content
+  form?.addEventListener("input", () => {
+    edited = true;
+    console.debug("editing detected, enabled beforeunload warning");
```

Comment:
The debug messages should be removed :-)

Reply:
Can we unregister the event listener after the first change? I find it somewhat unnecessary to trigger an event for every follow-up change :D

Reply:
I'm not sure, do the debug logs really harm? I mean it's not a `console.log()`, it is only shown when a user enables debug logging...

Reply:
Yes, its not harmful. But to me it is similar to print statements in Python code. Valid for testing stuff, but can be removed when done :shrug:

Reply:
Yeah, but we also have debug logging in python and do not remove it every time we solve an issue... It also makes the general debugging of problems easier in the future if all the debug logging is already there and you don't have to insert a lot of statements to track down a specific bug...